include/otbObiaLSBaatzSegmentationScheduler.txx

```diff
@@ -349,25 +389,42 @@ LSBaatzSegmentationScheduler<TInputImage>
 
     while( m_CurrentNumberOfIterations <= m_MaxNumberOfIterations &&
            accumulatedMemory > this->m_AvailableMemory &&
-           fusionSum > 0)
+           fusionSum > 0 && keepOn)
     {
         std::cout << "Current iteration " << m_CurrentNumberOfIterations << "/" << m_MaxNumberOfIterations << std::endl;
         std::cout << "Memory : " << accumulatedMemory << "/" << this->m_AvailableMemory << std::endl;
         std::cout << "Local sum : " << fusionSum << std::endl;
         //TODO : ExtractStabilityMargins();
-        
+
+	prevAccumulatedMemory = accumulatedMemory;
+
         AggregateStabilityMargins();
-        
         RunPartialSegmentation(accumulatedMemory, fusionSum);
-
         m_CurrentNumberOfIterations += m_PartialNumberOfIterations;
 
         //Extract for next iterations
         ExtractStabilityMargins();
 
+	//Test decreasing of accumulated Memory : not for first iteration
+	if (!firstIt)
+	{
+		double decreasing = (double)(prevAccumulatedMemory - accumulatedMemory)*100.0 / (double)prevAccumulatedMemory;
+		std::cout << "prevAccumulatedMemory : " << prevAccumulatedMemory << std::endl;
+		std::cout << "accumulatedMemory : " << accumulatedMemory << std::endl;
+		std::cout << "decreasing since last iteration : " << decreasing << std::endl;
+		if (decreasing < 1.0)
+		{
+			keepOn = false;
```

Comment:
If I understand correctly: we stop doing partial segmentation when the size of the graph does not decrease enough (with "enough" = less than 1%, defined line 415). This is related to the number of merged nodes VS the total number of nodes in the graph.
I agree with Julien this must be optional, and also settable : Maybe we could add a parameter group for the region merging stopping criterion, which would contain both the m_MaxNumberOfIterations and something like m_MinMergedNodesRatio.
I like this btw :+1: